backend/app/api/src/endpoints/global/organizations/SearchOrganizationEndpoint.ts

```diff
@@ -32,28 +33,45 @@ export class SearchOrganizationEndpoint extends Endpoint<Params, Query, Body, Re
 
     async handle(request: DecodedRequest<Params, Query, Body>) {
         // Escape query
-        const query = request.query.query.replace(/([-+><()~*"@\s]+)/g, ' ').replace(/[^\w\d]+$/, '');
-        if (query.length == 0) {
+        const query = request.query.query.replace(/([-+><()~*"@\s]+)/g, ' ').replace(/[^\w\d]+$/, '').trim();
+        if (query.length === 0) {
             // Do not try searching...
             return new Response([]);
         }
 
-        const match = {
-            sign: 'MATCH',
-            value: query + '*', // We replace special operators in boolean mode with spaces since special characters aren't indexed anyway
-            mode: 'BOOLEAN',
-        };
-
-        // We had to add an order by in the query to fix the limit. MySQL doesn't want to limit the results correctly if we don't explicitly sort the results on their relevance
-        const organizations = await Organization.where({ searchIndex: match, active: 1 }, {
-            limit: 15,
-            sort: [
-                {
-                    column: { searchIndex: match },
-                    direction: 'DESC',
-                },
-            ],
-        });
+        let matchValue: string;
+
+        if (query.includes(' ')) {
+            // give higher relevance if the searchindex includes the exact sentence
+            // give lower relevance if the last word is not a complete match
+            matchValue = `>("${query}") (${query}) <(${query}*)`;
+        }
+        else {
+            // give higher relevance if the searchindex includes the exact word
+            matchValue = `>${query} ${query}*`;
+        }
+
+        const limit = 15;
+
+        const whereMatch = SQL.where('searchIndex', SQLWhereSign.BooleanMatch, matchValue);
+
+        let organizations = await Organization.select()
+            .where(whereMatch)
+            .orderBy(whereMatch, 'DESC')
+            .limit(limit).fetch();
+
+        // if the limit is reached it is possible that organizations where the name starts with the query are missing -> fetch them and add them at the start
+        if (organizations.length === limit) {
+            const organizationsStartingWith = await Organization.select()
+                .where(new SQLWhereLike(SQL.column(Organization.table, 'searchIndex'), scalarToSQLExpression(`${query}%`)))
```

Comment:
```suggestion
                .where(new SQLWhereLike(SQL.column(Organization.table, 'name'), scalarToSQLExpression(`${query}%`)))
```
Moeten we hier niet dan de naam gebruiken ipv de search index?